PokemonAPI/PokemonAPI/Controllers/BaseController.cs

```diff
@@ -0,0 +1,13 @@
+﻿using Microsoft.AspNetCore.Mvc;
+
+namespace PokemonAPI.Controllers
+{
+    /// <summary>
+    /// Base controller implementation
+    /// </summary>
+    [ApiController]
+    [Route("[controller]/[action]")]
+    public abstract class BaseController : ControllerBase
+    {
+    }
```

Comment:
Base controller implementation that is an abstract class
We can put some base behaviors and properties here and all controllers implementing this will receive it.

The defining route as you see here will allow us to have a route configured for all controllers implementing this base controller. So instead of adding this attribute to every controller and hard-coding controller names, this will take care of it.